taichi/codegen/cuda/codegen_cuda.cpp

```diff
@@ -699,8 +733,8 @@ FunctionType CUDAModuleToFunctionConverter::convert(
     for (auto task : offloaded_tasks) {
       TI_TRACE("Launching kernel {}<<<{}, {}>>>", task.name, task.grid_dim,
                task.block_dim);
-      cuda_module->launch(task.name, task.grid_dim, task.block_dim, 0,
-                          {&context}, {});
+      cuda_module->launch(task.name, task.grid_dim, task.block_dim,
+                          task.dynamic_shared_array_bytes, {&context}, {});
```

Comment:
maybe I missed this in the code, why do we need to pass `dynamic_shared_array_bytes` to runtime?